### PR TITLE
add load config subsecion by key

### PR DESCRIPTION
**Pull request by lorentz-wu**
(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

What does this implement/fix? Explain your changes.
---------------------------------------------------
add load config with key

Does this close any currently open issues?
------------------------------------------
…

Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
…

### CODE REVIEW

**Comment by sherifabdlnaby** [comment]
This is a welcome addition, can you please add tests? also please delete the `go 1.15`  from `go.mod` because that makes the package not importable to older versions even if it is compatible (happened to me before tho it passes tests)

**Review by sherifabdlnaby — CHANGES_REQUESTED** [review]
This is a welcome addition, can you please add tests? also please delete the go 1.15 from go.mod because that makes the package not importable to older versions even if it is compatible (happened to me before tho it passes tests)

**Comment by lorentz-wu** [comment]
> This is a welcome addition, can you please add tests? also please delete the go 1.15 from go.mod because that makes the package not importable to older versions even if it is compatible (happened to me before tho it passes tests)

I removed go 1.15 from go.mod and add testcase LoadKey

File: load.go [hunk 1]
```diff
@@ -14,6 +14,15 @@ import (
 
 //Load load config into supported struct.
 func (c *Config) Load(configStruct interface{}) error {
+	return c.loadInternal("", configStruct)
+}
+
+//Load load config with key into supported struct.
```
Comment (MustafaSaber) [inline load.go:20]: I think you need to change this to `// LoadKey` for godocs.

**Pull request merged**